Test hardcopy with 'printmbcharset' to improve coverage

dpelle opened the pull request:
This PR improves test coverage of `:hardcopy` by 
testing with `'printmbcharset'`. 

It also speeds up test `Test_printoptions()` by limiting the range
of lines to print.

dpelle committed:
Test hardcopy with 'printmbcharset' to improve coverage

Also speed up test Test_printoptions() by limiting the range
of lines to print.

dpelle committed:
test_hardcopy was hanging in Windows CI

dpelle commented on src/testdir/test_hardcopy.vim:
```diff
@@ -63,6 +63,35 @@ func Test_printmbfont()
+  " FIXME: Unclear why this fails on Windows.
+  if !has('unix')
+    return
+  endif
```
The test was hanging on Window CI (Appveyor).
I merely disabled the new tests on Windows
as I don't have a Windows machine to reproduce it.

I'd appreciate if someone with a windows host could check
why this test is failing.  File `test_hardcopy.vim` has several
tests disabled on Windows:
* Test_printmbcharset()
* Test_printexpr()
* Test_errors()
* Test_empty_buffer()

vim-ml commented:
Hi Domnique,

On Sun, Mar 15, 2020 at 5:26 AM Dominique Pellé <vim-dev-github@256bit.org>
wrote:

> *@dpelle* commented on this pull request.
> ------------------------------
>
> In src/testdir/test_hardcopy.vim
> <https://github.com/vim/vim/pull/5783#discussion_r392668920>:
>
> > @@ -63,6 +63,35 @@ func Test_printmbfont()
>    bwipe
>  endfunc
>
> +func Test_printmbcharset()
> +  " FIXME: Unclear why this fails on Windows.
> +  if !has('unix')
> +    return
> +  endif
>
> The test was hanging on Window CI (Appveyor).
> I merely disabled the new tests on Windows
> as I don't have a Windows machine to reproduce it.
>
> I'd appreciate if someone with a windows host could check
> why this test is failing. File test_hardcopy.vim has several
> tests disabled on Windows:
>
>    - Test_printmbcharset()
>    - Test_printexpr()
>    - Test_errors()
>    - Test_empty_buffer()
>
>
The tests are failing on MS-Windows because "+postscript"
is missing. The Windows binary needs to be built with
"POSTSCRIPT=yes" in Appveyor.

I tested this on a MS-Windows machine with postscript enabled
and the above tests (after removing the unix check) are passing.

BTW, you may want to add the following modeline to test_hardcopy.vim:

" vim: shiftwidth=2 sts=2 expandtab

- Yegappan

dpelle committed:
Added missing checks for 'postscript' feature

dpelle committed:
Added POSTSCRIPT=yes for Windows + modeline in test_hardcopy.vim

dpelle commented:
Yegappan wrote;

> The tests are failing on MS-Windows because "+postscript" is missing.

Right, thank you.  I added has('postscript').
Also added POSTSCRIPT=yes for Windows and the suggested modeline.

vim-ml commented:
Hi,

On Sun, Mar 15, 2020 at 9:47 AM Dominique Pellé <vim-dev-github@256bit.org>
wrote:

> Yegappan wrote;
>
> The tests are failing on MS-Windows because "+postscript" is missing.
>
> Right, thank you. I added has('postscript').
> Also added POSTSCRIPT=yes for Windows and the suggested modeline.
>
>
> The Travis-CI huge/clang build for your PR failed when running the
test_termcodes test. This is not related to your change. I have added
a test for bringing up the popup menu using the mouse rightclick button.
That test is failing/hanging because the popupmenu is not dismissed
for some reason. In other test runs, this test is passing.

- Yegappan
